instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RoutesPreviewTest.kt

```diff
@@ -0,0 +1,139 @@
+package com.mapbox.navigation.instrumentation_tests.core
+
+import android.location.Location
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.trip.model.RouteProgressState
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.MapboxNavigationProvider
+import com.mapbox.navigation.core.directions.session.RoutesExtra
+import com.mapbox.navigation.core.directions.session.RoutesUpdatedResult
+import com.mapbox.navigation.core.preview.RoutesPreviewUpdate
+import com.mapbox.navigation.core.preview.RoutesPreviewUpdateReasons
+import com.mapbox.navigation.instrumentation_tests.activity.EmptyTestActivity
+import com.mapbox.navigation.instrumentation_tests.utils.MapboxNavigationRule
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.routeProgressUpdates
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.routesPreviewUpdates
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.routesUpdates
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.sdkTest
+import com.mapbox.navigation.instrumentation_tests.utils.routes.RoutesProvider
+import com.mapbox.navigation.instrumentation_tests.utils.routes.RoutesProvider.toNavigationRoutes
+import com.mapbox.navigation.testing.ui.BaseTest
+import com.mapbox.navigation.testing.ui.utils.getMapboxAccessTokenFromResources
+import com.mapbox.navigation.testing.ui.utils.runOnMainSync
+import kotlinx.coroutines.flow.first
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class RoutesPreviewTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.java) {
+
+    override fun setupMockLocation(): Location = mockLocationUpdatesRule.generateLocationUpdate {
+        latitude = 38.894721
+        longitude = -77.031991
+    }
+
+    @get:Rule
+    val mapboxNavigationRule = MapboxNavigationRule()
+    private lateinit var mapboxNavigation: MapboxNavigation
+
+    @Before
+    fun setUp() {
+        runOnMainSync {
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(activity)
+                    .accessToken(getMapboxAccessTokenFromResources(activity))
+                    .build()
+            )
+        }
+    }
+
+    @Test
+    fun transitions_free_drive_to_preview_to_active_guidance_to_free_drive() = sdkTest {
+        var currentRoutesPreview: RoutesPreviewUpdate? = null
```

Comment:
That's an interesting approach for verifying callback invocations. I've used the other 2 so far:
1) Like yours but with mutableList to keep track of all the values, not only the latest one (useful if you do the checks at the end of the test) - see `CopilotIntegrationTest`;
2) Channels (see `HistoryRecordingStateChangeObserverTest`);

I'd suggest trying channels here. This way you won't have to wait until data updates with flows and then separately check the current value. Because now it's a bit more difficult to keep in mind when to wait for what and there's a risk of introducing inconsistencies and flakiness. 
I mean that code like this:
```
mapboxNavigation.routesPreviewUpdates()
            .first { it.reason == RoutesPreviewUpdateReasons.PREVIEW_NEW }
        assertEquals(RoutesPreviewUpdateReasons.PREVIEW_NEW, currentRoutesPreview?.reason)
        assertEquals(routes, currentRoutesPreview!!.routesPreview!!.routesList)
```
can be reduced to sth like:
```
val routePreviewUpdate = channel.receive()
assertEquals(RoutesPreviewUpdateReasons.PREVIEW_NEW, croutePreviewUpdate.reason)
assertEquals(routes, routePreviewUpdate.routesPreview!!.routesList)
```
WDYT?

Reply:
main point of those listeners is to verify the case when callbacks weren't called https://github.com/mapbox/mapbox-navigation-android/pull/6495/files#diff-dd6f2bb7ffbeeccc5232ca194ae7b4ed2aecb887a9b87e497ab332e5d2b831c8R67

If I didn't need it at the beginning of the test I could have verify current state like this:
```
val currentRoutesPreivew = mapboxNavigation.routesPreviewUpdates()
            .first { it.reason == RoutesPreviewUpdateReasons.PREVIEW_NEW }
assertEquals(routes, currentRoutesPreview.routesPreview!!.routesList)
``` 
but as I already have `currentRoutesPreview` I just use it